### DISCUSSION TITLE
Cleanly reusing configuration between model Field()s and Query() parameters?

### DISCUSSION THREAD

**Discussion by acdha** [fastapi/fastapi#9713] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# query-string usage
async def get_translated_items(language: str | None = Query(  # noqa: B008
    default="en",
    min_length=2,
    max_length=2,
    description="A valid, 2-character ISO 639-1 language code",
    example="de",
)):
    pass

# POSTed JSON usage:

class SearchRequest(BaseModel):
    language: constr(min_length=2, max_length=2) | None = "en"

# response model
…
language: constr(min_length=2, max_length=2) | None = "en"
…
```

### Description

My API has some common values which are used in multiple places which I'd like to consolidate. The simplest example is the `language` code which for us means a two character ISO 639-1 code such as “en” or “zh”. We have some endpoints which use that as a query string parameter (`?language=es&…`), some which accept JSON objects which will have `"language": "es"` property in the appropriate location, and our response formats use that to denote the language for translated fields (e.g. `"titles": [{"language": "en", "value": "the English version of the title"}]`).

In practice, this means something like declaring separate `Field`, `Query`, etc. instances with the same validation, OpenAPI title/description/example, etc. values. Using standard Python practice that looks something like this:

```python
LANGUAGE_META = {
    "min_length": 2,
    "max_length": 2,
    "description": "A valid 2-character ISO 639-1 language code…",
    "example": "de",
}

LANGUAGE_FIELD = Field("en", **LANGUAGE_META)
LANGUAGE_FIELD_NO_DEFAULT = Field(None, **LANGUAGE_META)
LANGUAGE_QUERY = Query("en", **LANGUAGE_META)

LANGUAGE_T = constr(
    strict=True,
    **{k: LANGUAGE_META[k] for k in LANGUAGE_FIELD.get_constraints()},
)
```

That change for my views using query-string parameters looks like this:

```diff
-    language: str
-    | None = Query(  # noqa: B008
-        default="en",
-        min_length=2,
-        max_length=2,
-        description="A valid, 2-character ISO 639-1 language code",
-        example="de",
-    ),
+    language: LANGUAGE_T | None = LANGUAGE_QUERY,
```

… and the models for the POST views are like this:

```diff
-    language: constr(min_length=2, max_length=2) | None = "en"
+    language: LANGUAGE_T | None = LANGUAGE_FIELD
```

So that's not terrible but it still feels like more code than it should need. I could probably use Pydantic's custom data types to make a `LanguageCode(str)` class to subsume most of this, too, but that's trading more code to create it versus a cleaner interface (which is not a bad thing). It feels like there should be a cleaner way to do something like this where I really just want to tell FastAPI that there's a perfectly good `Field()` somewhere else and I'd like a `Query()` instance just like it. 

`__repr_args__()` happens to have exactly that data so I could do the revere (`LANGUAGE_FIELD = Field(**dict(LANGUAGE_QUERY.__repr_args__()))`) but I don't want to abuse an internal method like that, either. 

Did I miss something glaringly obvious in the documentation or is this something where it might be useful to be able to do something like `Query.from_field(MyModel.field)`?

Somewhat related, but only for Pydantic: https://github.com/pydantic/pydantic/discussions/4242

### Operating System

macOS

### FastAPI Version

0.95.2

### Python Version

Python 3.10.12

**Comment by YuriiMotov** [comment]
Tried to elaborate on this. Here are my intermediate results:
(I would use `Enum` instead of `str`, but here I kept it as it was)

```python
from typing import Annotated, TypeAlias
from fastapi import Body, FastAPI, Query

from pydantic import Field, StrictStr

app = FastAPI()


LANGUAGE_DESCR = "A valid 2-character ISO 639-1 language code…"
DEFAULT_LANGUAGE = "en"

Language: TypeAlias = Annotated[
    StrictStr | None,
    Field(
        min_length=2,
        max_length=2,
        description=LANGUAGE_DESCR,  # This doesn't take effect  in Swagger
        examples=["de"],
    ),
]

LanguageQuery: TypeAlias = Annotated[Language, Query(description=LANGUAGE_DESCR)]
LanguageBodyField: TypeAlias = Annotated[Language, Body(description=LANGUAGE_DESCR)]


@app.post("/")
async def post_(
    lng: LanguageQuery = DEFAULT_LANGUAGE,
    lng_2: Language = DEFAULT_LANGUAGE,  # This one will be without description in Swagger
    lng_body: LanguageBodyField = DEFAULT_LANGUAGE,
    lng_body_2: Annotated[  # This one will be without description in Swagger
        Language, Body()
    ] = DEFAULT_LANGUAGE,
):
    return {"lng": lng, "lng_2": lng_2, "lng_body": lng_body, "lng_body_2": lng_body_2}

```

I would expect it worked without `LanguageQuery` and `LanguageBodyField`. Just like this:
```python
async def post_(
    lng: Language = DEFAULT_LANGUAGE,
    lng_body: Annotated[Language, Body()] = DEFAULT_LANGUAGE,
):
```
but for now it will show schemas without `description`.
